src/game.cpp

```diff
@@ -2464,6 +2469,30 @@ void Game::processKeyInput()
 		toggleFast();
 	} else if (wasKeyDown(KeyType::NOCLIP)) {
 		toggleNoClip();
+	} else if (wasKeyDown(KeyType::MUTE)) {
+		float volume = g_settings->getFloat("sound_volume");
+		if(volume < 0.001) {
+			g_settings->setFloat("sound_volume", 1.0);
+			m_statustext = narrow_to_wide(gettext("Volume changed to 100%"));
+		} else {
+			g_settings->setFloat("sound_volume", 0.0);
+			m_statustext = narrow_to_wide(gettext("Volume changed to 0%"));
+		}
+		runData.statustext_time = 0;
+	} else if (wasKeyDown(KeyType::INC_VOLUME)) {
+		float new_volume = rangelim(g_settings->getFloat("sound_volume") + 0.1, 0, 1);
+		char buf[100];
+		g_settings->setFloat("sound_volume", new_volume);
+		sprintf(buf, gettext("Volume changed to %d%%"), myround(new_volume * 100));
```

Comment:
Nah, stringstream doesn't support placeholders and requires an additional copy.  However this should be using `snprintf` to prevent overflow (along with a more reasonably sized buffer) .